### PR TITLE
[incubator/vault] Add ability to configure liveness/readiness probe thresholds

### PR DESCRIPTION
**Pull request by haad**
#### What this PR does / why we need it:

Add ability to configure liveness/readiness probe thresholds and improves missing documentation.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ x] Chart Version bumped
- [ x] Variables are documented in the README.md
- [ x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: incubator/vault/templates/deployment.yaml [hunk 1]
```diff
@@ -129,6 +132,9 @@ spec:
             scheme: {{ if .Values.vault.config.listener.tcp.tls_disable -}}HTTP{{- else -}}HTTPS{{- end }}
           initialDelaySeconds: {{ .Values.vault.readiness.initialDelaySeconds }}
           periodSeconds: {{ .Values.vault.readiness.periodSeconds }}
+          failureThreshold: {{ .Values.vault.readiness.failureThreshold }}
+          successThreshold: {{ .Values.vault.readiness.successThreshold }}
+          timeoutSeconds: {{ .Values.vault.readiness.timeoutSeconds }}
```
Comment (jbialy) [inline incubator/vault/templates/deployment.yaml:137]: How about making these values the default unless overridden?

```suggestion
          failureThreshold: {{ .Values.vault.readiness.failureThreshold | default 3 }}
          successThreshold: {{ .Values.vault.readiness.successThreshold | default 1 }}
          timeoutSeconds: {{ .Values.vault.readiness.timeoutSeconds | default 1 }}
```

This way we only have to set these in the values.yaml when needed?

File: incubator/vault/templates/deployment.yaml [hunk 2]
```diff
@@ -118,6 +118,9 @@ spec:
             scheme: {{ if .Values.vault.config.listener.tcp.tls_disable -}}HTTP{{- else -}}HTTPS{{- end }}
           initialDelaySeconds: {{ .Values.vault.liveness.initialDelaySeconds }}
           periodSeconds: {{ .Values.vault.liveness.periodSeconds }}
+          failureThreshold: {{ .Values.vault.liveness.failureThreshold }}
+          successThreshold: {{ .Values.vault.liveness.successThreshold }}
+          timeoutSeconds: {{ .Values.vault.liveness.timeoutSeconds }}
```
Comment (jbialy) [inline incubator/vault/templates/deployment.yaml:123]: Similarly to the below:
```suggestion
          failureThreshold: {{ .Values.vault.liveness.failureThreshold | default 3 }}
          successThreshold: {{ .Values.vault.liveness.successThreshold | default 1 }}
          timeoutSeconds: {{ .Values.vault.liveness.timeoutSeconds | default 1 }}
```

File: incubator/vault/values.yaml [hunk 3]
```diff
@@ -238,16 +238,24 @@ vault:
     aliveIfSealed: true
     initialDelaySeconds: 30
     periodSeconds: 10
+    failureThreshold: 3
+    successThreshold: 1
+    timeoutSeconds: 1
```
Comment (jbialy) [inline incubator/vault/values.yaml:243]: These can now be removed 🤷 

```suggestion
```

File: incubator/vault/values.yaml [hunk 4]
```diff
@@ -238,16 +238,24 @@ vault:
...
   readiness:
     readyIfSealed: false
     readyIfStandby: true
     readyIfUninitialized: true
     initialDelaySeconds: 10
     periodSeconds: 10
+    failureThreshold: 3
+    successThreshold: 1
+    timeoutSeconds: 1
+
```
Comment (jbialy) [inline incubator/vault/values.yaml:253]:
```suggestion
```

**Comment by jbialy** [comment]
Hi @haad , thanks so much for the PR! Just a small suggestion. Please let me know what you think and if it makes sense. 😄

File: incubator/vault/templates/deployment.yaml (hunk 1 above)
Comment (jbialy) [reply incubator/vault/templates/deployment.yaml:137]: I realize that `vault.readiness.initialDelaySeconds` and vault.readiness.periodSeconds` should also follow the same convention and currently they don't! 😄 

Same goes for the `liveness` thresholds!

**Comment by haad** [comment]
IMO I don't like default variables in templates as then I have to search for them in 2 places instead of one (default values). If we don't want to keep them in default values file I can make them optional.

**Comment by jbialy** [comment]
> IMO I don't like default variables in templates as then I have to search for them in 2 places instead of one (default values). If we don't want to keep them in default values file I can make them optional.

I see your point. Agree. Let's go ahead and leave the new thresholds the way they are, defined in the default values! 👍

**Pull request merged**